dynawo/sources/Solvers/AlgebraicSolvers/DYNSolverKINAlgRestoration.cpp

```diff
@@ -66,6 +66,10 @@ SolverKINAlgRestoration::init(const shared_ptr<Model>& model, modeKin_t mode, do
   model_ = model;
   mode_ = mode;
 
+  // For some specific models, such as the dynamic line, the equation type could vary during the simulation.
+  model_->evalFType();
```

Comment:
A remark: There might be a performance overhead here. For now I guess it is ok, but later we might have to add a dedicated method (updateFType, updateYType) that does it only for the models that need it

Reply:
That's definitely a good point. I think you are right and already with large test cases and interesting events, there are chances that we get through this part a few times during each simulation.
I can create such a method at the model level and complete it with a default empty implementation everywhere else in a separate issue. I was hesitating to include it here as it is linked to the development but dealing with a more generic problem.

Do you think we could be able to create such a method from OMC outputs?

Reply:
I delete it at the moment and open it into #707